core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -160,8 +161,14 @@ class BaseUnit : INamed, IConstruction {
         if (uniqueTo != null && uniqueTo != civInfo.civName) return "Unique to $uniqueTo"
         if (civInfo.gameInfo.ruleSet.units.values.any { it.uniqueTo == civInfo.civName && it.replaces == name })
             return "Our unique unit replaces this"
-        if (!civInfo.gameInfo.gameParameters.nuclearWeaponsEnabled
-                && uniques.contains("Nuclear weapon")) return "Disabled by setting"
+        if (!civInfo.gameInfo.gameParameters.nuclearWeaponsEnabled && 
+            (
+                uniques.map{ it.getPlaceholderText()}.contains("Nuclear weapon of strength []") ||
+                // Deprecated since 3.15.3
+                    uniques.map{ it.getPlaceholderText()}.contains("Nuclear weapon")
+                //
```

Comment:
You're using this combo a lot, at least 4 times so far, I think it's OK to have BaseUnit.isNuclearWeapon()